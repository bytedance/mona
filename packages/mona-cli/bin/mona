#! /usr/bin/env node

const mona = require('../dist').default;
mona();
