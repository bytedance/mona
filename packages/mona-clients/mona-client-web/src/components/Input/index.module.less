.input {
  display: block;
  height: 1.4rem;
  text-overflow: clip;
  overflow: hidden;
  white-space: nowrap;
  font-family: PingFang SC, -apple-system, sans-serif;
  min-height: 1.4rem;
  -webkit-tap-highlight-color: transparent;
  background: transparent;
  border: 0;
  outline: 0;
}