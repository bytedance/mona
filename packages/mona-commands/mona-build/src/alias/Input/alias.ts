import { bubbleEventsAlias, basePropAlias, propAliasMap } from '../prop';
import { InputAlias as Alias } from '../type';

const alias: Alias = {
  ...basePropAlias,
  ...bubbleEventsAlias,
  // name: propAliasMap.name,
  value: propAliasMap.value,
  type: propAliasMap.type,
  password: propAliasMap.password,
  placeholder: propAliasMap.placeholder,
  disabled: propAliasMap.disabled,
  placeholderStyle: propAliasMap.placeholderStyle,
  maxLength: propAliasMap.maxLength,
  focus: propAliasMap.focus,
  cursorSpacing: propAliasMap.cursorSpacing,
  cursor: propAliasMap.cursor,
  selectionStart: propAliasMap.selectionStart,
  selectionEnd: propAliasMap.selectionEnd,
  onFocus: propAliasMap.onFocus,
  onBlur: propAliasMap.onBlur,
  onConfirm: propAliasMap.onConfirm,
  onInput: propAliasMap.onInput,
  adjustPosition: propAliasMap.adjustPosition,
  confirmType: propAliasMap.confirmType,
};

export default alias;
