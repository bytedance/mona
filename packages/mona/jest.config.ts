import baseConfig from '../../jestconfig.base';

module.exports = baseConfig;
