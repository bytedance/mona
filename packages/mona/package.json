{
  "name": "@bytedance/mona",
  "version": "0.2.0-alpha.9",
  "description": "Merchant app's developing and building tools",
  "main": "./dist/index",
  "scripts": {
    "build": "yarn clean && ttsc -b",
    "start": "ttsc -b -w",
    "clean": "rm -rf dist",
    "test": "jest"
  },
  "files": [
    "dist"
  ],
  "author": "zhangxiaowei.chris@bytedance.com",
  "license": "MIT",
  "publishConfig": {
    "access": "public",
    "registry": "https://registry.npmjs.org"
  },
  "devDependencies": {
    "copy-webpack-plugin": "^10.2.0",
    "react": "^17.0.2",
    "webpack": "^5.58.2",
    "webpack-chain": "^6.5.1"
  },
  "peerDependencies": {
    "react": "^17.0.2"
  },
  "gitHead": "cc5aa5283af595fd7d8a7b4921f54e2797504edd"
}
