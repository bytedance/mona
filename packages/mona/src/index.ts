export { createProjectConfig, ProjectConfig } from './createProjectConfig';
export { createAppConfig, AppConfig } from './createAppConfig';
export { createPageConfig, PageConfig } from './createPageConfig';
export { default as BaseApis } from './BaseApis';
export * from './BaseApis';
export { default as BaseComponents } from './BaseComponents';
export * from './BaseComponents';
export * from './types';
export * from './lifecycle';
