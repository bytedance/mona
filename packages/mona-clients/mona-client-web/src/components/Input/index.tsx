import React from 'react';
import { InputProps } from '@bytedance/mona';
import styles from './index.module.less';
import { useHandlers } from '../hooks';

const Input: React.FC<InputProps> = ({ children, ...restProps }) => {
  const { handleClassName, ...handlerProps} = useHandlers(restProps)
  
  return <input className={handleClassName(styles.input)} {...handlerProps} />
}

export default Input;
