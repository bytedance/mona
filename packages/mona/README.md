# @bytedance/mona
对外暴露提供构建时API
## 配置创建
### createProjectConfig
创建项目配置，`mona.config.ts`中使用
#### 例子
```js
// mona.config.ts
import { createProjectConfig } from '@bytedance/mona';
export default createProjectConfig({
  projectName: 'demo',
  input: './src/app.tsx',
  output: 'dist'
})
```

#### 参数说明
|  参数   | 说明  | 是否必填 | 类型 | 默认值 |
|  ----  | ----  | ---- | ---- | ---- |
| projectName  | 应用/插件名称 | 是 | string | - |
| input  | 入口文件路径 | 否 | string | - |
| output  | 打包后的文件目录路径 | 否 | string | - |
| raw  | 定义函数来修改webpack配置 | 否 | (options: any) => any | - |
| dev  | 用来修改本地服务配置 | 否 | string | { port: 9999 } |

### createAppConfig
创建应用配置，`app.config.ts`中使用
#### 例子
```js
// app.config.ts
import { createAppConfig } from '@bytedance/mona';
export default createAppConfig({
  pages: [
    'pages/Home/index',
    'pages/Info/index',
    'pages/List/index'
  ]
})
```
#### 参数说明
|  参数   | 说明  | 是否必填 | 类型 | 默认值 |
|  ----  | ----  | ---- | ---- | ---- |
| pages  | 页面路径 | 是 | string[] | - |

默认`pages`的第一项为首页
### createPageConfig
创建页面配置，需在每个页面目录的`page.config.ts`中使用

#### 例子
```js
// page.config.ts
import { createPageConfig } from '@bytedance/mona';
export default createPageConfig({
  navigationBarTitleText: 'page title'
})
```
#### 参数说明
|  参数   | 说明  | 是否必填 | 类型 | 默认值 |
|  ----  | ----  | ---- | ---- | ---- |
| navigationBarTitleText  | 当前页面标题 | 否 | string | - |
