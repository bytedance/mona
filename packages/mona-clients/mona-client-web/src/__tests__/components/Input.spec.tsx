import React from 'react';
import { mount } from 'enzyme';
import Input from '../../components/Input';
import mountTest from '../../../../../tests/shared/mountTest';
import handlerTest from '../../../../../tests/shared/handlerTest';
import { InputProps } from '@bytedance/mona';

describe('web component: Input', () => {
  mountTest(Input)
  handlerTest({ Component: Input, firstTag: 'input' });

  it('should render correctly', () => {
    expect(mount(<Input />).render()).toMatchSnapshot();
  })
  
  it('should have right properties', () => {
    const props: InputProps = {
      value: 'hello',
      type: 'number',
      password: true,
      placeholder: 'hello',
      placeholderStyle: { color: 'red' },
      disabled: true,
      maxLength: 100,
      focus: false,
      cursor: -1,
      selectionStart: -1,
      selectionEnd: -1,
      onInput: () => {},
      onFocus: () => {},
      onBlur: () => {},
      onConfirm: () => {},
      className: 'class-name',
      style: { color: 'red' },
    }
    const wrapper = mount(<Input {...props}>hello</Input>);
    expect(wrapper.render()).toMatchSnapshot();
  })

  
})
