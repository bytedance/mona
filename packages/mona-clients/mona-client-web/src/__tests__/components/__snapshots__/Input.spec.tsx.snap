// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`web component: Input should have right properties 1`] = `
LoadedCheerio {
  "0": <input
    class="input class-name  "
    disabled=""
    enterkeyhint="done"
    maxlength="100"
    name=""
    placeholder="hello"
    style="color: red;"
    type="password"
    value="hello"
  />,
  "_root": LoadedCheerio {
    "0": Document {
      "children": Array [
        <html>
          <head />
          <body />
        </html>,
      ],
      "endIndex": null,
      "next": null,
      "parent": null,
      "prev": null,
      "startIndex": null,
      "type": "root",
      "x-mode": "quirks",
    },
    "_root": [Circular],
    "length": 1,
    "options": Object {
      "decodeEntities": true,
      "xml": false,
    },
  },
  "length": 1,
  "options": Object {
    "decodeEntities": true,
    "xml": false,
  },
}
`;

exports[`web component: Input should render correctly 1`] = `
LoadedCheerio {
  "0": <input
    class="input   "
    enterkeyhint="done"
    maxlength="140"
    name=""
    type="text"
    value=""
  />,
  "_root": LoadedCheerio {
    "0": Document {
      "children": Array [
        <html>
          <head />
          <body />
        </html>,
      ],
      "endIndex": null,
      "next": null,
      "parent": null,
      "prev": null,
      "startIndex": null,
      "type": "root",
      "x-mode": "quirks",
    },
    "_root": [Circular],
    "length": 1,
    "options": Object {
      "decodeEntities": true,
      "xml": false,
    },
  },
  "length": 1,
  "options": Object {
    "decodeEntities": true,
    "xml": false,
  },
}
`;
